include/exadg/incompressible_navier_stokes/spatial_discretization/calculators/wall_shear_stress_calculator.cpp

```diff
@@ -0,0 +1,232 @@
+  double const                            kinematic_viscosity,
+  double const                            density)
+{
+  matrix_free = &matrix_free_in;
+  dof_index   = dof_index_in;
+  quad_index  = quad_index_in;
+
+  dynamic_viscosity = kinematic_viscosity * density;
+
+  // identify default matching Gauss-Lobatto point indices once up
+  // front since orientation of boundary faces *should be* constant
+  unsigned int const    fe_degree = matrix_free->get_dof_handler(dof_index).get_fe(0).degree;
+  dealii::FESystem<dim> fe_system(dealii::FE_DGQ<dim>(fe_degree), dim);
+  dealii::QGaussLobatto<dim - 1> face_quadrature(fe_degree + 1);
+  dealii::FEFaceValues<dim>      fe_face_values(fe_system,
+                                           face_quadrature,
+                                           dealii::update_quadrature_points);
+  dealii::QGaussLobatto<dim>     quadrature(fe_degree + 1);
+  dealii::FEValues<dim>          fe_values(fe_system, quadrature, dealii::update_quadrature_points);
+
+  face_to_cell_index.clear();
+  for(auto const & cell : matrix_free->get_dof_handler(dof_index).active_cell_iterators())
+  {
+    if(cell->is_locally_owned() == false)
+      continue;
+
+    fe_values.reinit(cell);
+    double const tol_sqrd = std::pow((cell->diameter() * rel_tol), 2);
+
+    face_to_cell_index.resize(cell->n_faces());
+
+    for(auto const face : cell->face_indices())
+    {
+      fe_face_values.reinit(cell, face);
+      std::vector<dealii::types::global_dof_index> face_cell_index_matches(
+        fe_face_values.n_quadrature_points, dealii::numbers::invalid_dof_index);
+
+      for(auto const i : fe_face_values.quadrature_point_indices())
+      {
+        dealii::Point<dim> const point_face = fe_face_values.quadrature_point(i);
+        for(auto const j : fe_values.quadrature_point_indices())
+        {
+          if(point_face.distance_square(fe_values.quadrature_point(j)) < tol_sqrd)
+          {
+            face_cell_index_matches[i] = j;
+            break;
+          }
+        }
+      }
+
+      face_to_cell_index[face] = face_cell_index_matches;
+    }
+
+    break;
+  }
+}
+
+template<int dim, typename Number>
+void
+WallShearStressCalculator<dim, Number>::compute_wall_shear_stress(
+  VectorType &                                  dst,
+  VectorType const &                            src,
+  std::shared_ptr<dealii::Mapping<dim> const>   mapping,
+  std::vector<dealii::types::boundary_id> const write_wall_shear_stress_on_IDs) const
+{
+  dst = 0;
+
+  // fill boundary ID to bool map OR write on all boundaries
+  AssertThrow(
+    write_wall_shear_stress_on_IDs.size() > 0,
+    dealii::ExcMessage(
+      "Provide at least write_wall_shear_stress_on_IDs = {dealii::numbers::invalid_boundary_id}.")) bool
+    write_on_all_boundary_IDs = false;
+  if(write_wall_shear_stress_on_IDs.size() == 1)
+  {
+    if(write_wall_shear_stress_on_IDs[0] == dealii::numbers::invalid_boundary_id)
+      write_on_all_boundary_IDs = true;
+  }
```

Comment:
I am against an implicit logic as realized here. I think this clarifies my original comment here https://github.com/exadg/exadg/pull/349/files#r1126058363. If we want to write on all boundaries, I think we have to list all of them in the application file.

From an application perspective, I would argue that no-slip wall boundaries (also referring to the picture you show in this PR) typically have the same boundary ID (the deal.II default value 0)?

Reply:
I went for default all based on `write_wall_shear_stress`, or selected ones when IDs are specified, is that ok for you as well?